### ISSUE TITLE
refactor: New file generates import from wrong module specifier

### ISSUE THREAD

**Issue by gsk143** [microsoft/TypeScript#58897]
Does this issue occur when all extensions are disabled?: Yes

Version: 1.90.0 (user setup)
Commit: 89de5a8d4d6205e5b11647eb6a74844ca23d2573
Date: 2024-06-04T19:33:54.889Z
Electron: 29.4.0
ElectronBuildId: 9593362
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Windows_NT x64 10.0.22631

Steps to Reproduce:
```js
import prisma from "@/prisma/client";
import { NextRequest, NextResponse } from "next/server";
import { z } from "zod";

const createIssueSchema = z.object({
  title: z.string().min(1).max(255),
  description: z.string().min(1),
});
```
Try refactoring createIssueSchema variable to a new file
``
2024-06-12 09:20:14.898 [error] <semantic> TypeScript Server Error (5.5.0-dev.20240505)
Debug Failure.
Error: Debug Failure.
    at Object.addImportFromExportedSymbol (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:154711:32)
    at c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:144661:19
    at Map.forEach (<anonymous>)
    at addTargetFileImports (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:144655:17)
    at getNewStatementsAndRemoveFromOldFile (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:143871:3)
    at doChange4 (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:144853:3)
    at c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:144842:77
    at _ChangeTracker.with (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:174963:5)
    at Object.getRefactorEditsToMoveToNewFile [as getEditsForAction] (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:144842:60)
    at Object.getEditsForRefactor (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:142944:31)
    at Object.getEditsForRefactor2 [as getEditsForRefactor] (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:150350:32)
    at IpcIOSession.getEditsForRefactor (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:191692:49)
    at getEditsForRefactor (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:189914:43)
    at c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:192078:69
    at IpcIOSession.executeWithRequestId (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:192070:14)
    at IpcIOSession.executeCommand (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:192078:29)
    at IpcIOSession.onMessage (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:192120:51)
    at process.<anonymous> (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\tsserver.js:528:14)
    at process.emit (node:events:514:28)
    at emit (node:internal/child_process:951:14)
    at process.processTicksAndRejections (node:internal/process/task_queues:83:21): Error: <semantic> TypeScript Server Error (5.5.0-dev.20240505)
Debug Failure.
Error: Debug Failure.
    at Object.addImportFromExportedSymbol (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:154711:32)
    at c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:144661:19
    at Map.forEach (<anonymous>)
    at addTargetFileImports (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:144655:17)
    at getNewStatementsAndRemoveFromOldFile (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:143871:3)
    at doChange4 (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:144853:3)
    at c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:144842:77
    at _ChangeTracker.with (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:174963:5)
    at Object.getRefactorEditsToMoveToNewFile [as getEditsForAction] (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:144842:60)
    at Object.getEditsForRefactor (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:142944:31)
    at Object.getEditsForRefactor2 [as getEditsForRefactor] (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:150350:32)
    at IpcIOSession.getEditsForRefactor (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:191692:49)
    at getEditsForRefactor (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:189914:43)
    at c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:192078:69
    at IpcIOSession.executeWithRequestId (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:192070:14)
    at IpcIOSession.executeCommand (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:192078:29)
    at IpcIOSession.onMessage (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\typescript.js:192120:51)
    at process.<anonymous> (c:\Users\ASUS\.vscode\extensions\ms-vscode.vscode-typescript-next-5.5.20240505\node_modules\typescript\lib\tsserver.js:528:14)
    at process.emit (node:events:514:28)
    at emit (node:internal/child_process:951:14)
    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)
	at n.create (c:\Users\ASUS\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\typescript-language-features\dist\extension.js:1:473136)
	at v.dispatchResponse (c:\Users\ASUS\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\typescript-language-features\dist\extension.js:1:467106)
	at v.dispatchMessage (c:\Users\ASUS\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\typescript-language-features\dist\extension.js:1:465950)
	at ChildProcess.<anonymous> (c:\Users\ASUS\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\typescript-language-features\dist\extension.js:1:465443)
	at ChildProcess.emit (node:events:514:28)
	at emit (node:internal/child_process:951:14)
	at process.processTicksAndRejections (node:internal/process/task_queues:83:21)
``
tsc version : 5.4.5

**Comment by lemomar** [comment]
I'm having this problem in a Next.js project as well, is it specific to Next?

**Comment by nolanleung** [comment]
Seems like the culprit is something in `^5.5` breaks refactoring imports from sym-linked packages. 


### Example (Works in v5.4.5)

Running `Refactor: Move to new file` on `SomeComponent` correctly moves it and the `Button` import to `SomeComponent.tsx`, and updates the import in `SomeOtherComponent.tsx` to `import { SomeComponent } from './SomeComponent'`.

```tsx
// apps/web/components/SomeOtherComponent.tsx

import { Button } from "@repo/ui";

export function SomeComponent() {
  return (
    <div>
      Some Component
      <Button>Click Me</Button>
    </div>
  );
}


export function SomeOtherComponent() {
  return (
    <div>
      <SomeComponent />
      <div>Some other component</div>
      <CheckList value={[]} options={[]} />
    </div>
  );
}


```

**Comment by etiennejcharles** [comment]
> I'm having this problem in a Next.js project as well, is it specific to Next?

No -  It's really based on the typescript version you're using for your project. 

If you do 

- `npm i typescript@5.4.5 -D`
- In your command pallette - `TypeScript: Select TypeScript Version..`
- Choose `5.4.5`

And try the command, you'll see that it works

**Comment by corned-beefhash** [comment]
> > I'm having this problem in a Next.js project as well, is it specific to Next?
> 
> No - It's caused by the typescript version you're using as commented by @nolanleung [ealier](https://github.com/microsoft/TypeScript/issues/58897#issuecomment-2212250416)
> 
> In your terminal:
> 
> * `npm i typescript@5.4.5 -D`
> * Then In your command pallette - `TypeScript: Select TypeScript Version..`
> * Choose `5.4.5`
> 
> And try the command, you'll see that it works ✅

this fixed my issue.  for anyone else


```
Debug Failure. False expression.
Error: Debug Failure. False expression.
    at getCompletionEntryCodeActionsAndSourceDisplay
```


i was getting this error when attempting to autoimport certain things.   TS server logs showed this 

Info 908  [17:52:51.447] response:
    `{"seq":0,"type":"response","command":"getCodeFixes","request_seq":64,"success":true,"body":[{"fixName":"import","description":"Add import from \"@tanstack/react-table\"","changes":[{"fileName":"[redacted]/column-select-button.tsx","textChanges":[{"start":{"line":3,"offset":1},"end":{"line":3,"offset":1},"newText":"import type { Table } from \"@tanstack/react-table\";\n"}]}]}]}`

but it wouldn't actually add any text to my file,  even though it showed the correct autoimport in the list.

which means the error was not from TS server?  But instead from vscode?    Just posting this incase it helps anyone

**Comment by divmgl** [comment]
Also really annoying that the Move to file and Move to new file has been swapped. I've been using Move to new file for years in the first position, why swap it?

**Comment by anback** [comment]
This really screwed up my flow for about to days, downgrading helped. see https://github.com/microsoft/TypeScript/issues/58897#issuecomment-2220808697

**Comment by Ife-Ody** [comment]
> * TypeScript: Select TypeScript Version

This worked for me! Typescript version 5.4.5

**Comment by vitonsky** [comment]
> In your terminal:
> 
>     * `npm i typescript@5.4.5 -D`
> 
>     * Then In your command pallette - `TypeScript: Select TypeScript Version..`
> 
>     * Choose `5.4.5`
> 
> 
> And try the command, you'll see that it works ✅

It works for me. If someone just faced with this problem - use this workaround

**Comment by sidouglas** [comment]
Does not work for me unfortunately. 
I added 

```
"typescript.tsdk": "/Users/simon/.nvm/versions/node/v21.5.0/lib/node_modules/typescript/lib",
```
[as per the docs](https://code.visualstudio.com/docs/typescript/typescript-compiling#_using-the-workspace-version-of-typescript)

output error when moving a Type declaration to a type.ts file.
```
<semantic> TypeScript Server Error (5.4.5) Debug Failure. False expression: No interactive refactor arguments available Error: Debug Failure. False expression: No interactive refactor arguments available at Object.getRefactorEditsToMoveToFile [as getEditsForAction] (/Users/simon/.nvm/versions/node/v21.5.0/lib/node_modules/typescript/lib/tsserver.js:140862:11) at Object.getEditsForRefactor (/Users/simon/.nvm/versions/node/v21.5.0/lib/node_modules/typescript/lib/tsserver.js:139733:31) at getEditsForRefactor2 (/Users/simon/.nvm/versions/node/v21.5.0/lib/node_modules/typescript/lib/tsserver.js:147297:32) at /Users/simon/.vscode/extensions/astro-build.astro-vscode-2.13.3-darwin-arm64/node_modules/astro-ts-plugin-bundle/index.js:5:1884 at Proxy.<anonymous> (/Users/simon/.vscode/extensions/vue.volar-2.0.28/node_modules/typescript-vue-plugin-bundle/index.js:5:1894) at IpcIOSession.getEditsForRefactor (/Users/simon/.nvm/versions/node/v21.5.0/lib/node_modules/typescript/l...
```

**Comment by KostkaBrukowa** [comment]
@sidouglas i guess in your case `@vue/typescript-plugin` plugin is the issue. I've had the same problem, and when I removed this plugin the refactor worked

**Comment by rsxdalv** [comment]
The workaround unfortunately doesn't work as smoothly if you don't have a node_modules in the current workspace root, i.e., if you have a monorepo with another sub folder.

I also tried the nightly version (via extension typescript nightly) 5.7 and it's broken too. If I do _move to file_ I at the very minimum least see an error. But trying to move to a new file anything bigger than a constant number fails.

**Comment by goncalossantos** [comment]
This was also happening to me due to the default VSCode version of TS, even though I was working without typescript on a js file. Doing the steps above and then changing the tsdk version fixed it for me

**Comment by iisaduan** [comment]
Minimal repro: 

```typescript
/// <reference path='fourslash.ts' />

// @Filename: /node_modules/zod/types.d.ts
//// declare const objectType: () => {};
//// 
//// export { objectType as object };

// @Filename: /node_modules/zod/external.d.ts
//// export * from "./types";

// @Filename: /node_modules/zod/index.d.ts
//// import * as z from "./external";
//// export * from "./external";
//// export { z };
//// export default z;

// @Filename: /test.ts
//// import { z } from "zod";
//// 
//// [|const createIssueSchema = z.object();|]

verify.moveToNewFile({
    newFileContents: {
        "/test.ts":
`
`,

        "/createIssueSchema.ts":
`import { z } from "zod";


const createIssueSchema = z.object();
`,
    },
});
```

`MoveToNewFile` should stop crashing after #60333, but the minimal repro will generate that `z` is imported from `zod/external` instead of just `z`

**Comment by laugri** [comment]
> * Then In your command pallette - `TypeScript: Select TypeScript Version..`

For those like me that don't understand why they don't have the entry showing up in their command palette, your focused tab has to be a .ts(x) file.

**Comment by magicdawn** [comment]
> > * Then In your command pallette - `TypeScript: Select TypeScript Version..`
> 
> For those like me that don't understand why they don't have the entry showing up in their command palette, your focused tab has to be a .ts(x) file.

This is another problem, I use "restart ts server" a lot, since problems with ts server can not auto recover, I have to manual restart.
I have to open a ts/tsx file then use the command palette to restart. 
why not showing these entries for whole workspace if ts server is running ? @The Team ???

**Comment by iisaduan** [comment]
@magicdawn since your crash still happens on 5.7, will you open a new issue with your repro? The case originally reported in this thread should no longer crash in 5.7, and I'd need to investigate your case separately

**Comment by aaronmw** [comment]
Putting this here for others, but I "resolved" this by realizing my own error. Granted, I do wish some kind of notification would have informed me of it... My code (example):

```typescript
function nonExportedFunction() {}

export function exportedFunction() {
  return nonExportedFunction()
}
```

Trying to move `exportedFunction` to a new file does nothing because, I assume, it doesn't want to just go and automatically export my `nonExportedFunction` so that the newly isolted `exportedFunction()` can import it.

Exported it myself, then "Move to new file" worked as expected `¯\_(ツ)_/¯ `

**Comment by iisaduan** [comment]
@aaronmw Your case seems to be a bug, since it is a case that _should_ work by generating `export` in front of `function nonExportedFunction() {}` when you invoke `move to new file`. Can you file a new issue with some more info about your case?

**Comment by MatiasGrimm** [comment]
This issue also persists in TypeScript version 5.8.2

```
[error] [vscode.typescript-language-features] provider FAILED
2025-04-25 07:59:18.960 [error] Error: <semantic> TypeScript Server Error (5.8.2)
Debug Failure. False expression: Expected symbol to be a module
Error: Debug Failure. False expression: Expected symbol to be a module
    at Object.addImportForModuleSymbol (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/typescript.js:157416:11)
    at /Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/typescript.js:147140:19
    at Map.forEach (<anonymous>)
    at addTargetFileImports (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/typescript.js:147133:17)
```

**Comment by scarf005** [comment]
issue persists on typescript 5.9.0: #247689

**Comment by ionutcatana** [comment]
It's present in typescript@latest (5.8.3)
```
<semantic> TypeScript Server Error (5.8.3)
Debug Failure. False expression: Expected symbol to be a module
Error: Debug Failure. False expression: Expected symbol to be a module
    at Object.addImportForModuleSymbol (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/typescript.js:157416:11)
    at /usr/share/code/resources/app/extensions/node_modules/typescript/lib/typescript.js:147140:19
    at Map.forEach (<anonymous>)
    at addTargetFileImports (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/typescript.js:147133:17)
    at getNewStatementsAndRemoveFromOldFile (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/typescript.js:146364:3)
    at doChange3 (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/typescript.js:146348:3)
    at /usr/share/code/resources/app/extensions/node_modules/typescript/lib/typescript.js:146333:79
    at _ChangeTracker.with (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/typescript.js:178070:5)
    at Object.getRefactorEditsToMoveToFile [as getEditsForAction] (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/typescript.js:146333:62)
    at Object.getEditsForRefactor (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/typescript.js:145454:31)
    at Object.getEditsForRefactor2 [as getEditsForRefactor] (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/typescript.js:152954:32)
    at IpcIOSession.getEditsForRefactor (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/typescript.js:195255:49)
    at getEditsForRefactor (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/typescript.js:193406:43)
    at /usr/share/code/resources/app/extensions/node_modules/typescript/lib/typescript.js:195676:15
    at IpcIOSession.executeWithRequestId (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/typescript.js:195665:14)
    at IpcIOSession.executeCommand (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/typescript.js:195674:29)
    at IpcIOSession.onMessage (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/typescript.js:195722:68)
    at process.<anonymous> (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/_tsserver.js:519:14)
    at process.emit (node:events:524:28)
    at emit (node:internal/child_process:950:14)
    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)
```

vscode version:
```
Version: 1.100.2
Commit: 848b80aeb52026648a8ff9f7c45a9b0a80641e2e
Date: 2025-05-14T21:47:40.416Z
Electron: 34.5.1
ElectronBuildId: 11369351
Chromium: 132.0.6834.210
Node.js: 20.19.0
V8: 13.2.152.41-electron.0
OS: Linux x64 6.12.22+bpo-amd64
```

**Comment by xmonkee** [comment]
Still present in Typescript v5.8.2 and v5.8.3

```
ERR <semantic> TypeScript Server Error (5.8.2)
Debug Failure. False expression: Expected symbol to be a module
Error: Debug Failure. False expression: Expected symbol to be a module
    at Object.addImportForModuleSymbol (/node_modules/typescript/lib/typescript.js:157416:11)
    at /node_modules/typescript/lib/typescript.js:147140:19
    at Map.forEach (<anonymous>)
    at addTargetFileImports (/node_modules/typescript/lib/typescript.js:147133:17)
    at getNewStatementsAndRemoveFromOldFile (/node_modules/typescript/lib/typescript.js:146364:3)
    at doChange4 (/node_modules/typescript/lib/typescript.js:147360:3)
    at /node_modules/typescript/lib/typescript.js:147349:77
    at _ChangeTracker.with (/node_modules/typescript/lib/typescript.js:178070:5)
    at Object.getRefactorEditsToMoveToNewFile [as getEditsForAction] (/node_modules/typescript/lib/typescript.js:147349:60)
    at Object.getEditsForRefactor (/node_modules/typescript/lib/typescript.js:145454:31)
    at Object.getEditsForRefactor2 [as getEditsForRefactor] (/node_modules/typescript/lib/typescript.js:152954:32)
    at IpcIOSession.getEditsForRefactor (/node_modules/typescript/lib/typescript.js:195255:49)
    at getEditsForRefactor (/node_modules/typescript/lib/typescript.js:193406:43)
    at /node_modules/typescript/lib/typescript.js:195676:15
    at IpcIOSession.executeWithRequestId (/node_modules/typescript/lib/typescript.js:195665:14)
    at IpcIOSession.executeCommand (/node_modules/typescript/lib/typescript.js:195674:29)
    at IpcIOSession.onMessage (/node_modules/typescript/lib/typescript.js:195722:68)
    at process.<anonymous> (/node_modules/typescript/lib/_tsserver.js:519:14)
    at process.emit (node:events:524:28)
    at emit (node:internal/child_process:950:14)
    at process.processTicksAndRejections (node:internal/process/task_queues:83:21): Error: <semantic> TypeScript Server Error (5.8.2)
Debug Failure. False expression: Expected symbol to be a module
Error: Debug Failure. False expression: Expected symbol to be a module
    at Object.addImportForModuleSymbol (/node_modules/typescript/lib/typescript.js:157416:11)
    at /node_modules/typescript/lib/typescript.js:147140:19
    at Map.forEach (<anonymous>)
    at addTargetFileImports (/node_modules/typescript/lib/typescript.js:147133:17)
    at getNewStatementsAndRemoveFromOldFile (/node_modules/typescript/lib/typescript.js:146364:3)
    at doChange4 (/node_modules/typescript/lib/typescript.js:147360:3)
    at /node_modules/typescript/lib/typescript.js:147349:77
    at _ChangeTracker.with (/node_modules/typescript/lib/typescript.js:178070:5)
    at Object.getRefactorEditsToMoveToNewFile [as getEditsForAction] (/node_modules/typescript/lib/typescript.js:147349:60)
    at Object.getEditsForRefactor (/node_modules/typescript/lib/typescript.js:145454:31)
    at Object.getEditsForRefactor2 [as getEditsForRefactor] (/node_modules/typescript/lib/typescript.js:152954:32)
    at IpcIOSession.getEditsForRefactor (/node_modules/typescript/lib/typescript.js:195255:49)
    at getEditsForRefactor (/node_modules/typescript/lib/typescript.js:193406:43)
    at /node_modules/typescript/lib/typescript.js:195676:15
    at IpcIOSession.executeWithRequestId (/node_modules/typescript/lib/typescript.js:195665:14)
    at IpcIOSession.executeCommand (/node_modules/typescript/lib/typescript.js:195674:29)
    at IpcIOSession.onMessage (/node_modules/typescript/lib/typescript.js:195722:68)
    at process.<anonymous> (/node_modules/typescript/lib/_tsserver.js:519:14)
    at process.emit (node:events:524:28)
    at emit (node:internal/child_process:950:14)
    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)
    at Function.create (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript-language-features/dist/extension.js:2:496515)
    at y.dispatchResponse (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript-language-features/dist/extension.js:2:490107)
    at y.dispatchMessage (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript-language-features/dist/extension.js:2:488548)
    at ChildProcess.<anonymous> (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript-language-features/dist/extension.js:2:488041)
    at ChildProcess.emit (node:events:524:28)
    at ChildProcess.emit (node:domain:489:12)
    at emit (node:internal/child_process:950:14)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
```

```
code --version
1.100.2
848b80aeb52026648a8ff9f7c45a9b0a80641e2e
arm64
```

**Comment by iisaduan** [comment]
@lukeramsden @ionutcatana @xmonkee Just from the stack trace, it seems like your cases have a different cause than the original bug in this issue. In order for us to investigate more, could you make a new issue with a repro?

**Comment by Cpt-Falcon** [comment]
Can confirm seeing this on v5.8.3 as well

**Comment by RyanCavanaugh** [comment]
Closing language service bugs related to the 6.0 implementation. For more information, see #62827
